ccl/changefeedccl: TestChangefeedOnlyInitialScan failed

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7605323?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7605323?buildTab=artifacts#/) on release-22.2 @ [bf7c1cf83275be4645fc77581be2cb9d5e8502c4](https://github.com/cockroachdb/cockroach/commits/bf7c1cf83275be4645fc77581be2cb9d5e8502c4):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan4217399105
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6350: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan4217399105
--- FAIL: TestChangefeedOnlyInitialScan (82.35s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6339: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3393/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6339
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (34.35s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6339: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3393/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6339
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (45.13s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (82.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21646

Epic CRDB-11732

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7637123?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7637123?buildTab=artifacts#/) on release-22.2 @ [2a71a5cca1f66313fd39956f1a3db070f67ade15](https://github.com/cockroachdb/cockroach/commits/2a71a5cca1f66313fd39956f1a3db070f67ade15):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan2393985084
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan2393985084
--- FAIL: TestChangefeedOnlyInitialScan (71.31s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4091/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (32.95s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4091/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (35.20s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (71.21s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7672519?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7672519?buildTab=artifacts#/) on release-22.2 @ [fe3ec49ff917b9b0eab2a4048fec31d56fc26daf](https://github.com/cockroachdb/cockroach/commits/fe3ec49ff917b9b0eab2a4048fec31d56fc26daf):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan2220115546
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan2220115546
--- FAIL: TestChangefeedOnlyInitialScan (88.35s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6339: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4091/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6339
        	Error:      	Received unexpected error:
        	            	still waiting for job status; current running
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*jobFeed).WaitForStatus.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:341
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:69
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:208
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:48
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.(*jobFeed).WaitForStatus
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:332
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestChangefeedOnlyInitialScan.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6339
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) still waiting for job status; current running
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (46.75s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (88.17s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4091/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (38.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7686015?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7686015?buildTab=artifacts#/) on release-22.2 @ [fd133ba3af3954c93584a501cf68dfa3131e586c](https://github.com/cockroachdb/cockroach/commits/fd133ba3af3954c93584a501cf68dfa3131e586c):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan2297327844
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan2297327844
--- FAIL: TestChangefeedOnlyInitialScan (73.74s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2816/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (35.33s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2816/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (35.16s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (73.62s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7698136?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7698136?buildTab=artifacts#/) on release-22.2 @ [fd133ba3af3954c93584a501cf68dfa3131e586c](https://github.com/cockroachdb/cockroach/commits/fd133ba3af3954c93584a501cf68dfa3131e586c):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan976783743
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan976783743
--- FAIL: TestChangefeedOnlyInitialScan (80.12s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3392/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (35.77s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3392/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (40.79s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (79.96s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7709785?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7709785?buildTab=artifacts#/) on release-22.2 @ [fd133ba3af3954c93584a501cf68dfa3131e586c](https://github.com/cockroachdb/cockroach/commits/fd133ba3af3954c93584a501cf68dfa3131e586c):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan646104949
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan646104949
--- FAIL: TestChangefeedOnlyInitialScan (91.07s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3204/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (40.78s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3204/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (45.96s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (90.99s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7721276?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7721276?buildTab=artifacts#/) on release-22.2 @ [fd133ba3af3954c93584a501cf68dfa3131e586c](https://github.com/cockroachdb/cockroach/commits/fd133ba3af3954c93584a501cf68dfa3131e586c):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan3069032946
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan3069032946
--- FAIL: TestChangefeedOnlyInitialScan (83.98s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3487/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (38.89s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3487/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (41.37s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (83.85s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScan [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7743163?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7743163?buildTab=artifacts#/) on release-22.2 @ [0ebce067f89a997f1daf04b4180d4dc5035c4fea](https://github.com/cockroachdb/cockroach/commits/0ebce067f89a997f1daf04b4180d4dc5035c4fea):


```
=== RUN   TestChangefeedOnlyInitialScan
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan1459350360
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestChangefeedOnlyInitialScan
    changefeed_test.go:6357: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/905465f697dc04c39f7d8a82cb057bfe/logTestChangefeedOnlyInitialScan1459350360
--- FAIL: TestChangefeedOnlyInitialScan (88.13s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3397/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_scan_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_scan_only (38.47s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
    changefeed_test.go:6346: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3397/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/changefeedccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/changefeedccl/changefeed_test.go:6346
        	Error:      	Not equal: 
        	            	expected: 5000
        	            	actual  : 0
        	Test:       	TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only
        --- FAIL: TestChangefeedOnlyInitialScan/enterprise/initial_backfill_only (46.25s)
=== RUN   TestChangefeedOnlyInitialScan/enterprise
    helpers_test.go:740: making server as secondary tenant
    helpers_test.go:830: making enterprise feed factory
    helpers_test.go:832: pgURL enterprise root
    --- FAIL: TestChangefeedOnlyInitialScan/enterprise (88.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92041 ccl/changefeedccl: TestChangefeedOnlyInitialScan failed [C-test-failure O-robot T-cdc branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScan.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -6527,6 +6527,12 @@ func TestChangefeedOnlyInitialScan(t *testing.T) {
 				sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY)`)
 				sqlDB.Exec(t, `INSERT INTO foo (a) SELECT * FROM generate_series(1, 5000);`)
 
+				// Most changefeed tests can afford to have a race condition between the initial
+				// inserts and starting the feed because the output looks the same for an initial
+				// scan and an insert. For tests with initial_scan=only, though, we can't start the feed
+				// until it's going to see all the initial inserts in the initial scan.
+				sqlDB.CheckQueryResultsRetry(t, `SELECT count(*) FROM foo`, [][]string{{`5000`}})
+
 				feed := feed(t, f, changefeedStmt)
 
 				sqlDB.Exec(t, "INSERT INTO foo VALUES (5005), (5007), (5009)")
@@ -6618,6 +6624,8 @@ func TestChangefeedOnlyInitialScanCSV(t *testing.T) {
 				sqlDB.Exec(t, "INSERT INTO foo VALUES (1, 'Alice'), (2, 'Bob'), (3, 'Carol')")
 				sqlDB.Exec(t, "INSERT INTO bar VALUES (1, 'a'), (2, 'b'), (3, 'c')")
 
+				sqlDB.CheckQueryResultsRetry(t, `SELECT count(*) FROM foo,bar`, [][]string{{`9`}})
+
 				feed := feed(t, f, testData.changefeedStmt)
 
 				sqlDB.Exec(t, "INSERT INTO foo VALUES (4, 'Doug'), (5, 'Elaine'), (6, 'Fred')")
@@ -6674,6 +6682,8 @@ func TestChangefeedOnlyInitialScanCSVSinkless(t *testing.T) {
 			t.Run(testName, func(t *testing.T) {
 				sqlDB.Exec(t, "CREATE TABLE foo (id INT PRIMARY KEY, name STRING)")
 				sqlDB.Exec(t, "INSERT INTO foo VALUES (1, 'Alice'), (2, 'Bob'), (3, 'Carol')")
+
+				sqlDB.CheckQueryResultsRetry(t, `SELECT count(*) FROM foo`, [][]string{{`3`}})
 
 				feed := feed(t, f, changefeedStmt)
 

